compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1181,13 +1181,37 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] w
    *  for equality would give the wrong result, so we should not use the sets
    *  for comparisons.
    */
```

Comment:
The comment above applies to the `canCompare` method so it should be moved down.